### ISSUE TITLE
Fugacity coefficients for pure substances is partially defect

### ISSUE THREAD

**Issue by halmi** [CoolProp/CoolProp#1373]
I've into a problem in calculating fugacity coefficients with the HEOS backend.

For my use case I'm calculating fugacity coefficients (amongst other properties) for a mixture of 3 gases (Hydrogen, Methane, CO2) at different compositions and states.

When a composition is defined so that only one component is present (e.g. [1,0,0]), the fugacity coefficients are calculated as NaN. (This does not happen with the REFPROP backend)

Changing the zero values slightly (e.g. to [1, 1e-20, 1e-20]), the fugacity coefficients are calculated correctly.

Additionally, if I only specify only one substance (e.g. just Hydrogen), then the fugacity calculation instantly leads to a segmentation fault. (This also does not happen with the REFPROP backend)

Interestingly, though, a case with two components (e.g. Hydrogen&Methane) with only one component present (composition [1,0]), the fugacity calculation works perfectly fine.

So, in short:
- Hydrogen&Methane&CO2 @ [1,0,0] --> NaN
- Hydrogen&Methane&CO2 @ [1,1e-20,1e-20] --> works fine
- Hydrogen&Methane @ [1,0] --> works fine
- Hydrogen --> segmentation fault

**Comment by ibell** [comment]
What version are you using?  I made some progress towards fixing that in this issue: #969 .

Also, please show your code, much easier to debug that way.

**Comment by halmi** [comment]
I'm using CoolProp 6.1.0 on macOS (installed via pip), I see that the relevant commit in #969 is more recent than that.

This gist shows the mentioned examples:
https://gist.github.com/halmi/8a5fc131eb14247606bb4bf35d75ac63

**Comment by ibell** [comment]
Can you try to use the updated binaries and let me know your experience?  You could install the development version with pip too..

**Comment by halmi** [comment]
I installed the latest development version via pip today and at the first test case the following changed:
```
st = CP.AbstractState('HEOS', 'Hydrogen&Methane&CO2')
st.specify_phase(CP.CoolProp.iphase_gas)
st.set_mole_fractions([1,0,0])
st.update(CP.PT_INPUTS, 50e5, 300)
phi = st.fugacity_coefficient(0)
print('phi 3c_a: {}'.format(phi))
```
already throws the following error `at st.update()`:
> Traceback (most recent call last):
  File "test.py", line 13, in <module>
    st.update(CP.PT_INPUTS, 50e5, 300)
  File "CoolProp/AbstractState.pyx", line 94, in CoolProp.CoolProp.AbstractState.update (CoolProp/CoolProp.cpp:20669)
  File "CoolProp/AbstractState.pyx", line 96, in CoolProp.CoolProp.AbstractState.update (CoolProp/CoolProp.cpp:20555)
ValueError: solver_rho_Tp was unable to find a solution for T=       300, p=     5e+06, with guess value    1952.24 with error: p is not a valid number

The other cases are unchanged:
- `Hydrogen&Methane&CO2` with `[1, 1e-20, 0]` works
- `Hydrogen&Methane` works
- `Hydrogen` leads to `segmentation fault`

**Comment by ibell** [comment]
For one, you are using hydrogen, so these problems may or may not be
related to the mole fractions you are using.  Try more similar fluids in
the meantime so that we can identify where the deficiencies are.

Basically, there are currently no guarantees that the mole fractions of
zero will work, though I made some attempts.  Clearly it was not completely
successful.  The short answer is to bump the mole fractions of all fluids
to greater than 0, even 1e-15 would do.  The better answer is to remove
fluids with zero mole fraction.  The best answer, obviously, is to fix
CoolProp, but that's pretty messy as I have learned.

**Comment by ibell** [comment]
The seg fault is the most worrying, it shouldn't do that at the very
least...
